### PR TITLE
Typescript type definitions for the public interface of the drand client

### PR DESCRIPTION
**Pull request by CluEleSsUK**

### CODE REVIEW

**Comment by CluEleSsUK** [comment]
This is my understanding of the public interface from looking through the README and tests - there were a few other classes I could have typed, but I don't think they're intended to be accessed directly

**Comment by CluEleSsUK** [comment]
In trying this out I've found a few shortcomings that I'm going to address

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    randomness: string
+    signature: string
+    previous_signature: string
+}
+
+export type NodeInfo = {
+    public_key: string
+    period: number
+    genesis_time: number
+    hash: string
```
Comment (AnomalRoil) [inline index.d.ts:13]: Is "hash" the "ChainHash"?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { AbortSignal } from "abort-controller"
+export type RandomnessBeacon = {
+    round: number
+    randomness: string
+    signature: string
+    previous_signature: string
```
Comment (AnomalRoil) [inline index.d.ts:6]: Notice that `previous_signature` is optional when using unchained randomness.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export interface Watcher {
+    watch(options: WatcherOptions): Promise<IterableIterator<RandomnessBeacon>>
+    close(): Promise<void>
+}
+
+export class HTTP {
+    static forURLs(urls: Array<string>, chainHash: string): Promise<Array<HTTP>>
+}
+
+export class Client {
```
Comment (AnomalRoil) [inline index.d.ts:43]: This is lacking a `chains` function to get the list of chainhashes like this: http://pl-us.testnet.drand.sh/chains

**Review by AnomalRoil — CHANGES_REQUESTED** [review]

**Comment by CluEleSsUK** [comment]
I'm going to close this PR, as I've done a full typescript migration on another branch

File: index.d.ts (hunk 3 above)
Comment (CluEleSsUK) [reply index.d.ts:43]: right now there isn't one in the js impl - I will raise an issue to add it though

File: index.d.ts (hunk 1 above)
Comment (CluEleSsUK) [reply index.d.ts:13]: yes - these keys are just what the go HTTP api returns

**Pull request closed without merging**